rustls/src/client/common.rs

```diff
@@ -38,12 +38,15 @@ impl<'a> ServerCertDetails<'a> {
 
 pub(super) struct ClientHelloDetails {
     pub(super) sent_extensions: Vec<ExtensionType>,
+    pub(super) extension_order_seed: u32,
 }
 
 impl ClientHelloDetails {
-    pub(super) fn new() -> Self {
+    pub(super) fn new(extension_order_seed: u16) -> Self {
         Self {
             sent_extensions: Vec::new(),
+            // leave bottom 16 bits clear for extension type.
+            extension_order_seed: (extension_order_seed as u32) << 16,
```

Comment:
Sorry, looking at this I have one more nit: maybe the field here should remain an `u16` and we shift it at use time (as in `(input.hello.extension_order_seed as u32) << 16`). What do you think? Seems like that's clearer about what we're doing here?

Reply:
I don't have a strong opinion about this, on one hand I think what you're suggesting is clearer, on the other hand this is a bit less processing if done ahead of time (although I doubt it'd matter xD)

Reply:
Changed it!